src/crimson/os/seastore/btree/fixed_kv_btree.h

```diff
@@ -502,9 +504,11 @@ class FixedKVBtree {
           &child_node);
       } else {
         if constexpr (leaf_has_children) {
-          ret = c.trans.get_extent(
-            i->get_val().paddr.maybe_relative_to(node->get_paddr()),
-            &child_node);
+          if (i->get_val().pladdr.is_paddr()) {
+            ret = c.trans.get_extent(
+              i->get_val().pladdr.get_paddr().maybe_relative_to(node->get_paddr()),
+              &child_node);
+          }
```

Comment:
Seems `ret` may not be assigned to any value in the else cases (509 and 510), and the followup logic to check the undefined `ret` will be undefined.

Early returns should be needed for the else cases.

Reply:
`check_node` is used for the lba pointer unit test in which pladdrs are all paddrs, I'll change the if clause to an assertion.

Reply:
Also handle the else case of `if constexpr (leaf_has_children)` properly? So there is no chance to misuse.